developersample/src/main/java/com/microsoft/identity/client/developersample/AuthUtil.java

```diff
@@ -0,0 +1,215 @@
+            AuthCallback callback = new AuthCallback(task, true);
+            mApplication.acquireToken(mActivity, SCOPES, callback);
+        }
+    }
+
+    // Do an silent login request
+    // Task is an handle to MainActivity which has overrriden useAccessToken and onRequestFailure
+    // On acquireToken success, useAccessToken is called with the AccessToken and on failure onRequestFailure is called
+    // with the exception
+    void doAcquireTokenSilent(final AuthenticatedTask task) {
+        AuthCallback callback = new AuthCallback(task, false);
+        mApplication.acquireTokenSilentAsync(SCOPES, mUsers.get(0), callback);
+    }
+
+    // Remove the user from the list of users being tracked by the application. This does not clear the cookies
+    void doSignout() {
+        if (getUserCount() > 0) {
+            mApplication.remove(mUsers.get(0));
+        }
+    }
+
+    // This is a shim, so MainActivity's onActivityResult can feed the result back to the PublicClientApplication's
+    // handleInteractiveRequestRedirect
+    // Any special handling of request or result code that the developer wants to do must be done here.
+    void doCallback(int requestCode, int resultCode, Intent data) {
+        mApplication.handleInteractiveRequestRedirect(requestCode, resultCode, data);
+    }
+
+    // This function just demonstrates how to request more scopes.
+    // The app does nothing with the newly acquired scopes yet.
+    void doExtraScopeRequest() {
+        mApplication.acquireToken(mActivity,
+                SCOPES,
+                "", // LoginHint
+                UiBehavior.SELECT_ACCOUNT,
+                "", // Extra query parameters
+                EXTRA_SCOPES,
+                "", // Authority
+                new AuthenticationCallback() {
+                    @Override
+                    public void onSuccess(AuthenticationResult authenticationResult) {
+                        // Not filled in for the sample app, ideally the callback should be passed and action taken on success
+                    }
+
+                    @Override
+                    public void onError(MsalException exception) {
+                        // Not filled in for the sample app, ideally the callback should be passed and error handled
+                    }
+
+                    @Override
+                    public void onCancel() {
+                        // Not filled in for the sample app
+                    }
+                });
+    }
+
+    // Get the list of users signed into the app. This app is designed to be a single user app, but if an app wants to
+    // handle more than one user, this is the way to get the list of users who are signed in.
+    private void getUsers() {
+        try {
+            mUsers = mApplication.getUsers();
+        } catch (final MsalException exc) {
+            Log.e(TAG, "Exception when getting users", exc);
+        }
+    }
+
+    class AuthCallback implements AuthenticationCallback {
```

Comment:
why static? made it private final

---
In reply to: [114901717](https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/165#discussion_r114901717) [](ancestors = 114901717)